integration_tests/test_pull_properties.py

```diff
@@ -53,9 +54,32 @@ def assert_expected_pull_state(self, project_dir):
 
 class AssetTrackingTestCase(integration_tests.TestCase):
 
+    def _create_git_repo(self, name):
+        def _call(cmd):
+            subprocess.check_call(cmd, stdout=subprocess.DEVNULL,
+                                  stderr=subprocess.DEVNULL)
+
+        def _call_with_output(cmd):
+            return subprocess.check_output(cmd).decode('utf-8').strip()
+
+        os.makedirs(name)
+        _call(['git', '-C', name, 'init'])
+        _call(['git', '-C', name, 'config',
+               '--local', 'user.name', 'Test User'])
+        _call(['git', '-C', name, 'config',
+               '--local', 'user.email', 'testuser@example.com'])
+        with open(os.path.join(name, 'testing'), 'w') as fp:
+            fp.write('testing')
+
+        _call(['git', '-C', name, 'add', 'testing'])
+        _call(['git', '-C', name, 'commit', '-am', 'testing'])
+        _call(['git', '-C', name, 'branch', 'feature'])
+        _call(['git', '-C', name, 'tag', 'feature-tag'])
+        return _call_with_output(['git', '-C', name, 'rev-parse', 'HEAD'])
```

Comment:
Having all of these pointing to the same commit seems like it's possible for the test to pass even if the hash-saving code saves the wrong hash. Do you think adding separate commits for master, feature, and feature-tag would make for more robust tests?